core/utils/dmljob.py

```diff
@@ -78,11 +88,14 @@ def deserialize_job(serialized_job):
         DMLJob: A DML Job object from serialized_job.
     """
     rest = serialized_job['job_data']
+    weights = None
+    if 'weights' in serialized_job and not serialized_job['weights'] is None:
+        weights = deserialize_weights(serialized_job['weights'])
```

Comment:
Why not just do:
```
weights = deserialize_weights(serialized_job.get['weights']) if serialized_job.get('weights', None) else None
```
At the very least, I think the if statement can be simplified to `if serialized_job.get('weights', None)`

Reply:
Not only do we need to check if 'weights' are in `serialized_job` but **also that the weights are not `None`**. This extra check was introduced by one of my commits of making serialized DMLJob always have the `weights` key in them for consistency.

Reply:
Looks good

Reply:
Don't think it's worth unresolving, but I believe both of my suggestions cover that @georgymh (`if serialized_job.get('weights', None)` will return `False` if the weights are not in `serialized_job` or if the weights are `None`.

Reply:
Hmm, you're right. Making the change.